examples/MQ/pixelSimSplit/src/devices/FairMQChunkMerger.cxx

```diff
@@ -0,0 +1,188 @@
+                    for ( int iobj = 0 ; iobj < arrayToAdd->GetEntries() ; ++iobj ) {
+                        FairMCTrack* temp = dynamic_cast<FairMCTrack*>(arrayToAdd->At(iobj));
+                        if ( temp->GetMotherId() >= 0 ) {
+                            temp->SetMotherId(temp->GetMotherId()+trackShift.back());
+                        }
+                    }
+                    tcaVector[iarray]->AbsorbObjects(arrayToAdd);
+                    //      LOG(info) << "FOUND ONE!, TCA has now " << tcaVector[iarray]->GetEntries() << " entries.";
+                }
+            }
+        }
+
+        for ( int iarray = 0 ; iarray < tcaVector.size() ; ++iarray ) {
+            if ( strcmp(tcaVector[iarray]->GetName(),"MCTrack") == 0 ) continue; // MCTrack already done, renumber all _other_ arrays...
+            //            LOG(info) << "BEFORE ADDING, TCA \"" << tcaVector[iarray]->GetName() << "\" has " << tcaVector[iarray]->GetEntries() << " entries.";
+            int addedArray = 0;
+            TClonesArray* arrayToAdd;
+ 
+            for ( MultiMapDef::iterator it = fRet.first ; it != fRet.second ; ++it ) {
+                if ( it->second.first == fMCSplitEventHeader->GetChunkStart() ) continue;
+                if ( strcmp(tcaVector[iarray]->GetName(),it->second.second->GetName()) == 0 ) {
+                    int objShift = trackShift[addedArray++];
+                    //     LOG(INFO) << "trying to add " << tcaVector[iarray]->GetName() << " and " << it->second.second->GetName() << "(shift = " << objShift << ")";
+                    arrayToAdd = dynamic_cast<TClonesArray*>(it->second.second);
+                    for ( int iobj = 0 ; iobj < arrayToAdd->GetEntries() ; ++iobj ) {
+                        FairMCPoint* temp = dynamic_cast<FairMCPoint*>(arrayToAdd->At(iobj));
+                        temp->SetTrackID(temp->GetTrackID()+objShift);
+                    }
+                }
+                tcaVector[iarray]->AbsorbObjects(arrayToAdd);
+                //                LOG(info) << "FOUND ONE!, TCA has now " << tcaVector[iarray]->GetEntries() << " entries.";
+            }
+        }
+        fObjectMap.erase(fRet.first,fRet.second);
+
+        FairMQParts partsOut;
+
+        fMCSplitEventHeader->SetNofChunks(1);
+        fMCSplitEventHeader->SetChunkStart(0);
+        
+        FairMQMessagePtr messEH(NewMessage());
+        Serialize<RootSerializer>(*messEH,fMCSplitEventHeader);
+        partsOut.AddPart(std::move(messEH));
+        
+        for ( int iarray = 0 ; iarray < tcaVector.size() ; ++iarray ) {
+            FairMQMessagePtr mess(NewMessage());
+            Serialize<RootSerializer>(*mess,tcaVector[iarray]);
+            partsOut.AddPart(std::move(mess));
+        }
+        //        LOG(info) << "created output message with " << partsOut.Size() << " parts.";
+        Send(partsOut, fOutputChannelName);
+        fNofSentMessages++;
+        if ( printInfo )
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Condition 'printInfo' is always false](https://app.codacy.com/app/FairRootGroup/FairRoot/pullRequest?prid=4139720)